jflex/src/main/java/jflex/core/AbstractLexScan.java

```diff
@@ -21,7 +21,9 @@
   int bufferSize = 16384;
 
   File file;
-  private Stack<File> files = new Stack<>();
+
+  @SuppressWarnings("JdkObsolete")
+  private final Stack<File> files = new Stack<>();
```

Comment:
I'm pretty sure `ArrayDeque` could offer the same features.

Reply:
Happy in principle to change out Stack for ArrayDeque.